Runtime/Scripts/AudioStreamTrack.cs

```diff
@@ -52,133 +52,70 @@ public class AudioStreamTrack : MediaStreamTrack
-                    {
-                        return (int)(m_renderPos + m_samplesForBuffering - BufferPosition);
-                    }
-                }
-
-                return 0;
-            }
-        }
-
+        public AudioSource Renderer => _streamRenderer.Source;
 
         internal class AudioStreamRenderer : IDisposable
         {
-            private bool m_bufferReady = false;
-            private readonly Queue<float[]> m_recvBufs = new Queue<float[]>();
-            private AudioBufferTracker m_bufInfo;
-            public event OnAudioReceived OnAudioReceived;
             private bool disposed;
 
             internal IntPtr self;
 
-            private AudioSource m_audioSource;
+            private AudioSource _audioSource;
+            private AudioCustomFilter _filter;
+
+
 
             public AudioSource Source 
             {
                 get
                 {
-                    return m_audioSource;
+                    return _audioSource;
                 }
                 set
                 {
-                    m_audioSource = value;
+                    _audioSource = value;
+                    AddFilter(_audioSource);
                 }
             }
 
-            private bool IsSameParams(int sampleRate, int channels)
-            {
-                return Source.clip.samples == sampleRate &&
-                    Source.clip.channels == channels;
-            }
 
-            private void UpdateParams(int sampleRate, int channels)
+            private static T GetOrAddComponent<T>(GameObject go) where T : Component
             {
-                var isPlaying = m_audioSource.isPlaying;
-
-                // Replace AudioClip for updating parameter
-                AudioClip oldClip = m_audioSource.clip;
-                UnityEngine.Object.DestroyImmediate(oldClip);
-                m_audioSource.clip =
-                    CreateClip($"{m_audioSource.name}-{GetHashCode():x}", sampleRate, channels);
-
-                // Restart AudioSource
-                if (isPlaying)
-                    m_audioSource.Play();
+                T comp = go.GetComponent<T>();
+                if (!comp)
+                    comp = go.AddComponent<T>();
+                return comp;
             }
 
-            static AudioClip CreateClip(string clipName, int sampleRate, int channels)
+            private void AddFilter(AudioSource source)
             {
-                int lengthSamples = sampleRate;  // sample length for 1 second
-                return AudioClip.Create(
-                    clipName, lengthSamples, channels, sampleRate, false);
+                if (_filter != null)
+                    return;
+                _filter = GetOrAddComponent<AudioCustomFilter>(source.gameObject);
+                _filter.hideFlags = HideFlags.HideInInspector;
+                _filter.onAudioRead += SetData;
+                source.Play();
+                Debug.Log("AddFilter");
```

Comment:
I think no need this log in package script

Reply:
fixed it